### ISSUE TITLE
Some characters are not serialized correctly.

### ISSUE THREAD

**Issue by JFreakDK** [okta/okta-sdk-java#934]
### Describe the bug?

The payload is not serialised correctly if it contains characters like “æ”, “ø” and “å”.

Could the issue be caused by the way the StringEntity is initialised: https://github.com/okta/okta-sdk-java/blob/4da9773be99c0c8bcb380ce988672afd1fe0d476/api/src/main/resources/custom_templates/ApiClient.mustache#L832C53-L832C53

The ISO-8859-1 charset is used in the StringEntity constructor that is used.

### What is expected to happen?

I guess the payload should be serialised with the UTF-8 charset.

### What is the actual behavior?

The payload is serialised into something that Okta rejects

### Reproduction Steps?

This should be enough to reproduce the issue:
```
        UserBuilder.instance().setFirstName("ÆØÅæøå").buildAndCreate(userApi);
```

The response I get is:
```
ApiException{code=400, responseHeaders={...}, responseBody='{"errorCode":"E0000003","errorSummary":"The request body was not well-formed.","errorLink":"E0000003","errorId":"...","errorCauses":[]}
'}
	at com.okta.sdk.resource.client.ApiClient.processResponse(ApiClient.java:957)
	at com.okta.sdk.resource.client.ApiClient.invokeAPI(ApiClient.java:1109)
	at com.okta.sdk.resource.api.UserApi.createUser(UserApi.java:394)
	at com.okta.sdk.resource.api.UserApi.createUser(UserApi.java:339)
	at com.okta.sdk.impl.resource.DefaultUserBuilder.buildAndCreate(DefaultUserBuilder.java:469)
	at com.trackunit.operator.UserProfileProd.testAdditionalPropertiesForUserProfile(UserProfileProd.java:49)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

```

### Java Version

openjdk version "20.0.2" 2023-07-18
OpenJDK Runtime Environment Temurin-20.0.2+9 (build 20.0.2+9)
OpenJDK 64-Bit Server VM Temurin-20.0.2+9 (build 20.0.2+9, mixed mode, sharing)

### SDK Version

12.0.2 and 13.0.0

**Comment by JFreakDK** [comment]
Maybe it would be a fix to change the default header to `application/json; charset=utf-8` in `com.okta.sdk.resource.client.ApiClient#selectHeaderContentType`:
https://github.com/okta/okta-sdk-java/blob/4da9773be99c0c8bcb380ce988672afd1fe0d476/api/src/main/resources/custom_templates/ApiClient.mustache#L753C41-L753C47

`org.apache.hc.core5.http.io.entity.StringEntity` defaults to `StandardCharsets.ISO_8859_1` if no charset is provided.

**Comment by arvindkrishnakumar-okta** [comment]
Fix released in 13.0.2
